Automaton/Features/DateWithDestiny.cs

```diff
@@ -307,12 +460,90 @@ private void TargetAndMoveToEnemy(IGameObject target)
+                    if (DistanceToTarget() < 10)
+                    {
+                        Svc.Log.Debug("Distance within 10");
+                        if (P.Lifestream.CanChangeInstance())
+                        {
+                            var numberOfInstances = P.Lifestream.GetNumberOfInstances();
+                            if (SuccessiveInstanceChanges < numberOfInstances - 1)
+                            {
+                                Svc.Log.Debug("Changing instances.");
+                                SuccessiveInstanceChanges += 1;
+                                var nextInstance = ((P.Lifestream.GetCurrentInstance() + 1) % numberOfInstances) + 1; // instances are 1-indexed
+                                P.Lifestream.ChangeInstance(nextInstance);
+                            }
+                            else
+                            {
+                                SuccessiveInstanceChanges = 0;
+                                // wait 10 seconds before trying again to prevent instance change spam
+                                System.Threading.Thread.Sleep(10000);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        Svc.Log.Debug("Not close enough to change instance. Moving closer");
+                        // interact distance is between 8 and 10. less than 8 and you will run into the base of the aetheryte
+                        var closerToAetheryte = Svc.Targets.Target.Position - (Vector3.Normalize(Svc.Targets.Target.Position - Player.Position) * 8);
+                        P.Navmesh.PathfindAndMoveTo(closerToAetheryte, false);
+                    }
+                }
+            }
+            else // not targeting an aetheryte
+            {
+                var closestAetheryteDataId = Coords.GetNearestAetheryte((int)Player.Territory, Player.Position);
+                var closestAetheryte = Svc.Objects
+                    .Where(x => x is { ObjectKind: ObjectKind.Aetheryte })
+                    .FirstOrDefault(x => x.DataId == closestAetheryteDataId);
+                if (Coords.GetDistanceToAetheryte(closestAetheryteDataId, Player.Position) < _distanceToTargetAetheryte)
+                {
+                    TeleportLock = 0;
+                    Svc.Targets.Target = closestAetheryte;
+                }
+                else
+                {
+                    if (TeleportLock == 0) // ensures only one teleport command goes through
+                    {
+                        TeleportLock = closestAetheryteDataId;
+                        Svc.Log.Debug("Teleporting to nearby aetheryte: " + closestAetheryteDataId);
+                        Coords.TeleportToAetheryte(closestAetheryteDataId);
+                    }
+                    else
+                    {
+                        Svc.Log.Debug("Teleporting is locked.");
```

Comment:
with the amount of logic now I think it might be best if this feature was reworked into a state system